### PR TITLE
save model with best metric in training

### LINKED ISSUE #42: [FEATURE] Pointer to checkpoint with best [param]
**Describe the solution you'd like**
Add a configuration where user can get information to checkpoint with best selected attributes ( example : best training loss, best selected validation metrics, etc )

**Describe alternatives you've considered**
Add this function at the training pipeline and dump the information into relevan log file or possibly to the experiment logger

### PR DESCRIPTION
**Pull request by triwahyuu**
## Type of changes

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

close #42 

## What is the new behavior?

- adding save_best_metrics in config
- save checkpoint model with best metric from the save_best_metrics config
- not saving model on hyperparameter optimization

## Checklist

- [ ] Unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have filled the `[Unreleased]` [Changelog](https://github.com/nodefluxio/vortex/blob/master/CHANGELOG.md)

### CODE REVIEW

**Comment by alphinside** [comment]
lgtm, updated docs, please review @triwahyuu

**Comment by triwahyuu** [comment]
LGTM, ready for merge @alphinside
